### ISSUE TITLE
Container is vulnerable to multiple attacks

### ISSUE THREAD

**Issue by Lofter1** [tfenster/mssql-image#4]
I ran docker scan on an image I created using this image as base, and it returned the following output:

```
Tested 42 dependencies for known issues, found 10 issues.


Issues to fix by upgrading:

  Upgrade commons-codec:commons-codec@1.4 to commons-codec:commons-codec@1.13 to fix
  ✗ Information Exposure [Low Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCODEC-561518] in commons-codec:commons-codec@1.4
    introduced by commons-codec:commons-codec@1.4

  Upgrade commons-collections:commons-collections@3.2.1 to commons-collections:commons-collections@3.2.2 to fix
  ✗ Deserialization of Untrusted Data [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-472711] in commons-collections:commons-collections@3.2.1
    introduced by commons-collections:commons-collections@3.2.1
  ✗ Deserialization of Untrusted Data [Critical Severity][https://snyk.io/vuln/SNYK-JAVA-COMMONSCOLLECTIONS-30078] in commons-collections:commons-collections@3.2.1
    introduced by commons-collections:commons-collections@3.2.1

  Upgrade org.apache.hive:hive-exec@2.1.0 to org.apache.hive:hive-exec@2.3.4 to fix
  ✗ Arbitrary Files Access [Low Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHIVE-460760] in org.apache.hive:hive-exec@2.1.0
    introduced by org.apache.hive:hive-exec@2.1.0
  ✗ Access Restriction Bypass [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHIVE-72580] in org.apache.hive:hive-exec@2.1.0
    introduced by org.apache.hive:hive-exec@2.1.0
  ✗ Authentication Bypass [High Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHIVE-72581] in org.apache.hive:hive-exec@2.1.0
    introduced by org.apache.hive:hive-exec@2.1.0


Issues with no direct upgrade or patch:
  ✗ Information Disclosure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-1015415] in com.google.guava:guava@14.0.1
    introduced by org.apache.hive:hive-exec@2.1.0 > com.google.guava:guava@14.0.1
  This issue was fixed in versions: 30.0-android, 30.0-jre
  ✗ Deserialization of Untrusted Data [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236] in com.google.guava:guava@14.0.1
    introduced by org.apache.hive:hive-exec@2.1.0 > com.google.guava:guava@14.0.1
  This issue was fixed in versions: 24.1.1-android, 24.1.1-jre
  ✗ Integer Overflow [High Severity][https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-173761] in com.google.protobuf:protobuf-java@2.5.0
    introduced by org.apache.hive:hive-exec@2.1.0 > com.google.protobuf:protobuf-java@2.5.0
  This issue was fixed in versions: 3.4.0
  ✗ Information Exposure [Medium Severity][https://snyk.io/vuln/SNYK-JAVA-ORGAPACHEHIVE-30638] in org.apache.hive:hive-common@2.1.0
    introduced by org.apache.hive:hive-exec@2.1.0 > org.apache.hive:hive-common@2.1.0
  This issue was fixed in versions: 2.2.0
```

**Comment by tfenster** [comment]
@Lofter1 thanks for bringing this up. Unfortunately, I don't see how I could fix this as I am only installing the standard base and CU installers from Microsoft

**Comment by Lofter1** [comment]
Well, for one you could update to 2022 ltsc instead of 2019 ltsc (I could create a PR for that if you want), maybe that fixes some vulnerabilities. 

Other than that, is there a specific reason why you use the dotnet framework image? Instead of, for example
[windows server](https://hub.docker.com/_/microsoft-windows-server)
[windows server core](https://hub.docker.com/_/microsoft-windows-servercore)
[nano server](https://hub.docker.com/_/microsoft-windows-nanoserver)

**Comment by tfenster** [comment]
@Lofter1 Because of https://github.com/microsoft/Windows-Containers/issues/182, I don't have a Github runner on Windows Server 2022 currently, so I can't build 2022 images right now. As soon as that is fixed, I will create 2022 images as well. Adding it will just be an additional string in the array [here](https://github.com/tfenster/mssql-image/blob/35d582ab8d7dfe64fcdb1a936caf67adc16d7463/.github/workflows/build-images-on-tag.yml#L24), so that won't be a problem. 

For the images:

- Windows Server: Just a bigger image including additional features I don't need
- Windows Server Core without .NET framework: Last time I checked, SQL Server had the .NET framework as prereq, so I just went with the image.
- Nano Server: I don't think SQL Server supports running on Nano

But maybe most importantly: The base image (mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-...) doesn't have those vulnerabilities, so I am pretty sure they come with the SQL install. So no matter in which way I change the base image, I would guess that the vulnerabilities will be the same when I install SQL

```
PS C:\Users\tfenster> docker scan  mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-2004

Testing mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-2004...

Package manager:   linux
Project name:      docker-image|mcr.microsoft.com/dotnet/framework/runtime
Docker image:      mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-2004
Platform:          windows/amd64

✔ Tested mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-2004 for known vulnerabilities, no vulnerable paths found.

Note that we do not currently have vulnerability data for your image.

For more free scans that keep your images secure, sign up to Snyk at https://dockr.ly/3ePqVcp
```
